From 0d70acea641c71d8185dcb8fc60363aea1699db0 Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Tue, 11 Mar 2025 12:53:32 +0200
Subject: [PATCH] [SW-190031] boot_desc: prepare for moving to cmn_lib

There are multiple boot_desc implementations in embedded
project, each with its own variants.
Instead, it is better to maintain one library for all.
This patch prepares for using boot_desc in emb_lib_cmn.

Change-Id: I2d6dc35c16ab1e8830d8cfa921b7c6fa81c0f585
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/lib/common/boot_desc.c b/lib/common/boot_desc.c
index ee3a3d6..b231ba8 100644
--- a/lib/common/boot_desc.c
+++ b/lib/common/boot_desc.c
@@ -25,7 +25,8 @@
 static u8 desc_buf[BOOT_DESC_MAX_IMGS * IMG_DESC_SZ];
 static bool desc_initialized;
 
-static void boot_desc_print_magic(char *name, struct hl_fw_boot_magic *magic)
+/* before removal and using emb_lib_common instead */
+static void __boot_desc_print_magic(char *name, struct hl_fw_boot_magic *magic)
 {
 	int i;
 
@@ -43,7 +44,7 @@
 
 static bool boot_desc_img_desc_is_valid(struct hl_fw_img_desc *img_desc)
 {
-	boot_desc_print_magic("FW IMG DESC", &img_desc->magic);
+	__boot_desc_print_magic("FW IMG DESC", &img_desc->magic);
 
 	if ((img_desc->magic.val & 0xFFFFFF00) != HL_FW_IMG_DESC_MAGIC)
 		return false;
@@ -127,7 +128,7 @@
 		if (boot_ptrs.boot_ptrs_sz > BOOT_PTRS_SZ)
 			preboot_puts("Warning: boot_ptrs size is bigger than expected\n");
 	}
-	boot_desc_print_magic("FW boot PTRs", &boot_ptrs.magic);
+	__boot_desc_print_magic("FW boot PTRs", &boot_ptrs.magic);
 
 	if (boot_ptrs.magic.val != HL_FW_BOOT_MAGIC)
 		return 0;
